cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java

```diff
@@ -308,25 +308,31 @@ private void writeKafkaPartitionState(StringBuilder sb, SortedSet<PartitionInfo>
   private void populateKafkaBrokerLogDirState(Map<Integer, Set<String>> onlineLogDirsByBrokerId,
                                               Map<Integer, Set<String>> offlineLogDirsByBrokerId,
                                               Set<Integer> brokers)
-      throws ExecutionException, InterruptedException, TimeoutException {
+      throws ExecutionException, InterruptedException {
     Map<Integer, KafkaFuture<Map<String, LogDirInfo>>> logDirsByBrokerId = describeLogDirs(brokers, _adminClientConfigs).values();
 
     for (Map.Entry<Integer, KafkaFuture<Map<String, LogDirInfo>>> entry : logDirsByBrokerId.entrySet()) {
       onlineLogDirsByBrokerId.put(entry.getKey(), new HashSet<>());
       offlineLogDirsByBrokerId.put(entry.getKey(), new HashSet<>());
 
-      entry.getValue().get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((key, value) -> {
-        if (value.error == Errors.NONE) {
-          onlineLogDirsByBrokerId.get(entry.getKey()).add(key);
-        } else {
-          offlineLogDirsByBrokerId.get(entry.getKey()).add(key);
-        }
-      });
+      try {
+        entry.getValue().get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((key, value) -> {
+          if (value.error == Errors.NONE) {
+            onlineLogDirsByBrokerId.get(entry.getKey()).add(key);
+          } else {
+            offlineLogDirsByBrokerId.get(entry.getKey()).add(key);
+          }
+        });
+      } catch (TimeoutException te) {
```

Comment:
If it is observed whenever a broker is dead, then I believe we should handle it differently. In the current scenario, for each dead broker we would spend `LOGDIR_RESPONSE_TIMEOUT_MS`, which seems high.

If we observe this whenever a broker is dead, can we check whether `replicaCountByBrokerId` and `offlineReplicaCountByBrokerId` have the same size for a broker before attempting to retrieve the corresponding `logDir` info?